### DISCUSSION TITLE
Safe to pass different input selectors from different slices?

### DISCUSSION THREAD

**Discussion by n-ii-ma** [reactjs/reselect#590] [Q&A] [answered]
I was wondering if it's safe to pass two input selectors in `createSelector` where each selector belongs to different slices?

If it is actually safe, then would it be better to create the selector instance in one of those slices or in the component where it is being used?

**Comment by markerikson** [comment, marked as answer]
Sure, there's no problem there.  In fact, that's part of the point of selectors in the first place - to encapsulate looking up date from different parts of the state.

We typically recommend defining selectors in a slice file, but it's up to you - Redux doesn't care where the selector function actually lives.

See https://redux.js.org/usage/deriving-data-selectors for more details.
